File: 1сем/ОАиП лабы/Common7/IDE/ProjectTemplates/CSharp/Windows Root/Windows/1049/WPFCustomControl/CustomControl1.cs
using System;
using System.Collections.Generic;
$if$ ($targetframeworkversion$ >= 3.5)using System.Linq;
$endif$using System.Text;
$if$ ($targetframeworkversion$ >= 4.5)using System.Threading.Tasks;
$endif$using System.Windows;
using System.Windows.Controls;
using System.Windows.Data;
using System.Windows.Documents;
using System.Windows.Input;
using System.Windows.Media;
using System.Windows.Media.Imaging;
using System.Windows.Navigation;
using System.Windows.Shapes;

namespace $safeprojectname$
{
    /// <summary>
    /// Выполните шаги 1a или 1b, а затем 2, чтобы использовать этот настраиваемый элемент управления в файле XAML.
    ///
    /// Шаг 1a. Использование настраиваемого элемента управления в файле XAML, существующем в текущем проекте.
    /// Добавьте атрибут XmlNamespace к корневому элементу файла разметки, где он 
    /// должен использоваться:
    ///
    ///     xmlns:MyNamespace="clr-namespace:$safeprojectname$"
    ///
    ///
    /// Шаг 1b. Использование этого настраиваемого элемента управления в файле XAML, существующем в текущем проекте.
    /// Добавьте атрибут XmlNamespace к корневому элементу файла разметки, где он 
    /// должен использоваться:
    ///
    ///     xmlns:MyNamespace="clr-namespace:$safeprojectname$;assembly=$safeprojectname$"
    ///
    /// Потребуется также добавить ссылку на проект из проекта, в котором находится файл XAML
    /// в данный проект и пересобрать во избежание ошибок компиляции:
    ///
    ///     Правой кнопкой мыши щелкните проект в обозревателе решений и выберите команду
    ///     "Добавить ссылку"->"Проекты"->[Выберите этот проект]
    ///
    ///
    /// Шаг 2)
    /// Продолжайте дальше и используйте элемент управления в файле XAML.
    ///
    ///     <MyNamespace:$safeitemrootname$/>
    ///
    /// </summary>
    public class $safeitemrootname$ : Control
    {
        static $safeitemrootname$()
        {
            DefaultStyleKeyProperty.OverrideMetadata(typeof($safeitemrootname$), new FrameworkPropertyMetadata(typeof($safeitemrootname$)));
        }
    }
}

File: 1сем/ОАиП лабы/Common7/IDE/ItemTemplatesCache/VisualBasic/WPF/1049/WPFCustomControl/CustomControl1.vb
' Выполните шаги 1a или 1b, а затем 2, чтобы использовать этот пользовательский элемент управления в файле XAML.
'
' Шаг 1a. Использование пользовательского элемента управления в файле XAML, существующем в текущем проекте.
' Добавьте атрибут XmlNamespace в корневой элемент файла разметки, где он 
' будет использоваться:
'
'     xmlns:MyNamespace="clr-namespace:$rootnamespace$"
'
'
' Шаг 1б. Использование пользовательского элемента управления в файле XAML, существующем в другом проекте.
' Добавьте атрибут XmlNamespace в корневой элемент файла разметки, где он 
' будет использоваться:
'
'     xmlns:MyNamespace="clr-namespace:$rootnamespace$;assembly=$rootnamespace$"
'
' Потребуется также добавить ссылку из проекта, в котором находится файл XAML,
' на данный проект и заново выполнить построение во избежание ошибок компиляции:
'
'     Щелкните правой кнопкой мыши нужный проект в обозревателе решений и выберите
'     "Добавить ссылку"->"Проекты"->[Поиск и выбор проекта]
'
'
' Шаг 2)
' Теперь можно использовать элемент управления в файле XAML. Внимание! Технология Intellisense
' в редакторе XML пока не поддерживает пользовательские элементы управления и их дочерние элементы.
'
'     <MyNamespace:$safeitemrootname$/>
'

Imports System.Windows.Controls.Primitives


Public Class $safeitemrootname$
    Inherits System.Windows.Controls.Control

    Shared Sub New()
        'Вызов OverrideMetadata сообщает системе, что этот элемент содержит стиль, отличающийся от стиля в базовом классе.
        'Стиль определен в файле themes\generic.xaml
        DefaultStyleKeyProperty.OverrideMetadata(GetType($safeitemrootname$), new FrameworkPropertyMetadata(GetType($safeitemrootname$)))
    End Sub

End Class
